packages/Filter/src/components/Item/Item.js

```diff
@@ -40,18 +40,25 @@ function Item(props) {
     value,
     renderValueField: renderCustomValueField,
   } = props;
+
   const { columns, data, filterRef, onChangeOperator, operator, rulesByType } = React.useContext(FilterContext);
   const I18n = useI18n();
 
   const selectedColumnType = columns.find(({ id }) => id === selectedColumnId).type;
 
-  const selectOptions = React.useMemo(() => {
+  const uniqueSelectOptions = React.useMemo(() => {
     switch (selectedColumnType) {
       case types.columnTypes.SINGLE_SELECT:
       case types.columnTypes.MULTI_SELECT:
-        return data.filter(
-          (obj, index, arr) => arr.map(datum => datum[selectedColumnId]).indexOf(obj[selectedColumnId]) === index
-        );
+        return data.filter((row, index, dataOriginal) => {
+          const selectedColumnValues = dataOriginal.map(datum => datum[selectedColumnId]); // [{id, label}, {id, label}, ...]
+          const position = selectedColumnValues.findIndex(
+            selectedColumnValue =>
+              selectedColumnValue.id === row[selectedColumnId].id &&
+              selectedColumnValue.label === row[selectedColumnId].label
+          );
+          return position === index;
+        });
```

Comment:
Should we just use https://lodash.com/docs/4.17.15#uniqWith so the next person won't be confused 😂